### PR TITLE
📦 NEW: Pipe v1 support

### PR DESCRIPTION
**Pull request by asharirfan**
This PR adds the support of Pipe version 1 to BaseAI core and server packages. The new version looks as follows:

```ts
const pipe = {
	name: 'summary',
	description: 'AI Summary agent',
	status: 'private',
	model: 'openai:gpt-4o-mini',
	stream: true,
	json: false,
	store: true,
	moderate: true,
	top_p: 1,
	max_tokens: 100,
	temperature: 0.7,
	presence_penalty: 1,
	frequency_penalty: 1,
	stop: [],
	tool_choice: 'auto',
	parallel_tool_calls: false,
	messages: [
		{
			role: 'system',
			content: `You are a helpful AI assistant. Make everything Less wordy.`,
		},
	],
	variables: [],
	tools: [],
	memory: []
}
```

### CODE REVIEW

File: packages/baseai/src/deploy/index.ts [hunk 1]
```diff
@@ -1033,34 +1029,26 @@ async function uploadDocument(signedUrl: string, document: Blob) {
...
 	memoryName: string;
 	documentName?: string;
 }) {
-	const isOrgAccount = account.apiKey.includes(':');
-	const ownerLogin = isOrgAccount
-		? account.apiKey.split(':')[0]
-		: account.login;
-	const baseUrl = `https://api.langbase.com/beta`;
+	// Base URL
 	const baseUrlV1 = `https://api.langbase.com/v1`;
```
Comment (ahmadbilaldev) [inline packages/baseai/src/deploy/index.ts:1039]: Shouldn't we just update baseUrl value instead of naming it v1?

**Review by ahmadbilaldev — APPROVED** [review]
Left a comment but overall looks good to me.

File: packages/baseai/src/dev/routes/v1/pipes/run.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	json: z.boolean().optional(),
+	store: z.boolean().optional(),
+	moderate: z.boolean().optional(),
+	top_p: z.number().optional(),
+	max_tokens: z.number().optional(),
+	temperature: z.number().optional(),
+	presence_penalty: z.number().optional(),
+	frequency_penalty: z.number().optional(),
+	stop: z.array(z.string()).optional(),
+	tool_choice: z.string(),
```
Comment (msaaddev) [inline packages/baseai/src/dev/routes/v1/pipes/run.ts:26]: The `tool_choice` is not correct here. It is the following:

```ts
interface ToolChoiceFunction {
	type: 'function';
	function: ToolFunction;
}

export type ToolChoice = 'auto' | 'required' | 'none' | ToolChoiceFunction;
```

File: packages/baseai/src/dev/routes/v1/pipes/run.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+	parallel_tool_calls: z.boolean(),
+	messages: z.array(schemaMessage),
+	tools: z.array(z.unknown()).default([]),
+	memory: z.array(z.object({ name: z.string().trim().min(1) })).default([]),
+	variables: VariablesSchema
+});
+
+export type Pipe = z.infer<typeof PipeSchema>;
```
Comment (msaaddev) [inline packages/baseai/src/dev/routes/v1/pipes/run.ts:34]: We have a correct Pipe type present inside `packages/baseai/types/pipes.ts`. Why not use from there instead?

File: packages/baseai/src/dev/llms/call-fireworks.ts [hunk 4]
```diff
@@ -30,7 +31,8 @@ export async function callFireworks({
 		dlog('Fireworks request params', transformedRequestParams);
 
 		// Fireworks llama-3.1 405b behaves weirdly with stop value. Bug on their side. Omitting it.
-		if (pipe.model.name === 'llama-v3p1-405b-instruct')
+		const modelName = pipe.model.split(':')[1];
```
Comment (msaaddev) [inline packages/baseai/src/dev/llms/call-fireworks.ts:34]: I think `modelParams.model` will also have the model name.

File: packages/baseai/src/dev/llms/utils.ts [hunk 5]
```diff
@@ -17,10 +18,11 @@ export function handleLlmError({
 	});
 }
 
-export function applyJsonModeIfEnabled(modelParams: ModelParams, pipe: any) {
+export function applyJsonModeIfEnabled(modelParams: ModelParams, pipe: Pipe) {
+	const currentModel = pipe.model.split(':')[1];
```
Comment (msaaddev) [inline packages/baseai/src/dev/llms/utils.ts:22]: I think `modelParams.model` will already have this name.

File: packages/baseai/src/dev/llms/utils.ts [hunk 6]
```diff
@@ -30,11 +32,12 @@ export function applyJsonModeIfEnabled(modelParams: ModelParams, pipe: any) {
 
 export function applyJsonModeIfEnabledForGoogle(
 	transformedRequestParams: any,
-	pipe: any
+	pipe: Pipe
 ) {
+	const currentModel = pipe.model.split(':')[1];
```
Comment (msaaddev) [inline packages/baseai/src/dev/llms/utils.ts:37]: Same. I think `modelParams.model` will already have this name

File: packages/baseai/src/dev/routes/v1/pipes/run.ts [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+	if (error instanceof ApiErrorZod) {
+		throw error;
+	}
+
+	const errorMessage =
+		error instanceof Error
+			? error.message
+			: 'Unexpected error occurred in beta/generate';
+
+	dlog('Error beta/generate.ts:', error);
```
Comment (msaaddev) [inline packages/baseai/src/dev/routes/v1/pipes/run.ts:98]: I think the dlogs here are not correct.

File: packages/baseai/src/dev/utils/thread/get-few-shot-messages.ts [hunk 8]
```diff
@@ -1,4 +1,4 @@
-import type { Pipe } from '@/dev/routes/beta/pipes/run';
+import type { Pipe } from '@/dev/routes/v1/pipes/run';
```
Comment (msaaddev) [inline packages/baseai/src/dev/utils/thread/get-few-shot-messages.ts:1]: Same. We should keep one Pipe type. The one in `packages/baseai/types/pipe.ts` is the correct one.

File: packages/core/src/utils/to-old-pipe-format.ts [hunk 9]
```diff
@@ -1,27 +1,5 @@
...
-type Provider =
-	| typeof OPEN_AI
-	| typeof ANTHROPIC
-	| typeof TOGETHER_AI
-	| typeof GOOGLE
-	| typeof GROQ
-	| typeof COHERE
-	| typeof FIREWORKS_AI
-	| typeof PERPLEXITY;
+import {getProvider} from './get-provider';
```
Comment (msaaddev) [inline packages/core/src/utils/to-old-pipe-format.ts:2]: Do we need this file anymore?

Comment (asharirfan) [reply packages/core/src/utils/to-old-pipe-format.ts:2]: No, it isn't being used anywhere.

File: packages/baseai/src/dev/llms/call-google.ts [hunk 10]
```diff
@@ -48,15 +49,29 @@ export async function callGoogle({
 }
 
 function buildModelParams(
```
[the remark this answers is not in the document]
Comment (asharirfan) [reply packages/baseai/src/dev/llms/call-google.ts:51]: Agreed. We can cover this in another PR.

File: packages/baseai/src/dev/llms/utils.ts (hunk 6 above)
Comment (asharirfan) [reply packages/baseai/src/dev/llms/utils.ts:37]: `modelParams` is not being passed to this function

**Review by msaaddev — APPROVED** [review]
LGTM. Please first release and test a snapshot version before merging to `main`.

**Pull request closed without merging**